GoogleSignIn/Sources/GIDAuthorizationResponseHelper.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+@class GIDAuthFlow;
+@class GIDConfiguration;
+@class OIDAuthorizationResponse;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// A list of potential current flow names.
+typedef NS_ENUM(NSInteger, GIDFlowName) {
+/// The Sign In flow.
+  SignIn,
+/// The Verify flow.
+  Verify,
+};
+
+/// A helper class to process the authorization response.
+@interface GIDAuthorizationResponseHelper : NSObject
+
+/// The authorization response to process.
+@property(nonatomic, readonly) OIDAuthorizationResponse *authorizationResponse;
+
+/// The EMM support version.
+@property(nonatomic, readwrite, nullable) NSString *emmSupport;
+
+/// The name of the current flow.
+@property(nonatomic, readonly)GIDFlowName flowName;
+
+/// The configuration of the current flow.
+@property(nonatomic, readwrite) GIDConfiguration *configuration;
+
+/// Initializes a new instance of the `GIDAuthorizationResponseHelper` class with the provided fields.
+///
+/// @param authorizationResponse The authorization response to be processed.
+/// @param emmSupport The EMM support version.
+/// @param flowName The name of the current flow.
+/// @param configuration The configuration.
+/// @return A new initialized instance of the `GIDAuthorizationResponseHelper` class.
+- (instancetype)
+    initWithAuthorizationResponse:(nullable OIDAuthorizationResponse *)authorizationResponse
```

Comment:
Makes sense, added! In regard to simplifying the impl code, do you mind clarifying your vision on what could change?

Reply:
I believe you check for whether or not there is an `_authorizationResponse` somewhere in the impl. You won't need that if it's non-null.

Reply:
Actually ignore my suggestion to not check: `if (_authorizationResponse) { ...`. We should still have the param marked `nonnull`, but the check in the impl is needed since the nullability annotation does not guarantee that the passed argument is `nonnull`.

Reply:
Sounds good! Thanks for double checking.